sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/_shared/async_client_base.py

```diff
@@ -2,83 +2,135 @@
+        )
+    return ImplClient(credentials=credential, pipeline=pipeline)
+
+def _import_models(api_version: str):
+    if api_version == V7_0_VERSION:
+        from ._generated.v7_0 import models as impl_models
+    elif api_version == V2016_10_01_VERSION:
+        from ._generated.v2016_10_01 import models as impl_models
+    else:
+        raise NotImplementedError(
+            "API version {} is not supported by this package. ".format(api_version)
+            + "Supported versions are {} and {}".format(V7_0_VERSION, V2016_10_01_VERSION)
+        )
+    return impl_models
+
+
+class AsyncKeyVaultClientBase(object):
     """Base class for async Key Vault clients"""
 
-    @staticmethod
-    def _create_config(credential: "TokenCredential", api_version: str = None, **kwargs: "Any") -> Configuration:
-        if api_version is None:
-            api_version = KeyVaultClient.DEFAULT_API_VERSION
-        config = KeyVaultClient.get_configuration_class(api_version, aio=True)(credential, **kwargs)
-        config.authentication_policy = AsyncChallengeAuthPolicy(credential)
-
-        # replace the autorest-generated UserAgentPolicy and its hard-coded user agent
-        # https://github.com/Azure/azure-sdk-for-python/issues/6637
-        config.user_agent_policy = UserAgentPolicy(base_user_agent=USER_AGENT, **kwargs)
-
-        # Override config policies if found in kwargs
-        # TODO: should be unnecessary after next regeneration (written 2019-08-02)
-        if "user_agent_policy" in kwargs:
-            config.user_agent_policy = kwargs["user_agent_policy"]
-        if "headers_policy" in kwargs:
-            config.headers_policy = kwargs["headers_policy"]
-        if "proxy_policy" in kwargs:
-            config.proxy_policy = kwargs["proxy_policy"]
-        if "logging_policy" in kwargs:
-            config.logging_policy = kwargs["logging_policy"]
-        if "retry_policy" in kwargs:
-            config.retry_policy = kwargs["retry_policy"]
-        if "custom_hook_policy" in kwargs:
-            config.custom_hook_policy = kwargs["custom_hook_policy"]
-        if "redirect_policy" in kwargs:
-            config.redirect_policy = kwargs["redirect_policy"]
-
-        return config
-
-    def __init__(self, vault_url: str, credential: "TokenCredential", **kwargs: "Any") -> None:
+    def __init__(self, vault_url: str, credential: "AsyncTokenCredential", **kwargs: "Any") -> None:
         if not credential:
             raise ValueError(
-                "credential should be an object supporting the TokenCredential protocol, "
+                "credential should be an object supporting the AsyncTokenCredential protocol, "
                 "such as a credential from azure-identity"
             )
         if not vault_url:
             raise ValueError("vault_url must be the URL of an Azure Key Vault")
 
         self._vault_url = vault_url.strip(" /")
-
+        api_version = kwargs.pop("api_version", None) or DEFAULT_API_VERSION
```

Comment:
not kwargs.pop("api_version", DEFAULT_API_VERSION)?

Reply:
```py
>>> {"api_version":None}.pop("api_version", "some default") == "some default"
False
```